userspace/libsinsp/filter.cpp

```diff
@@ -2040,30 +2044,64 @@ void sinsp_evttype_filter::add(string &name,
 			filters->push_back(wrap);
 		}
 	}
+
+	for(auto &tag: tags)
+	{
+		auto it = m_filter_by_tag.find(tag);
+
+		if(it == m_filter_by_tag.end())
+		{
+			m_filter_by_tag.emplace(std::make_pair(tag, std::list<filter_wrapper*>()));
+			it = m_filter_by_tag.find(tag);
+		}
+
+		it->second.push_back(wrap);
+	}
 }
 
-void sinsp_evttype_filter::enable(string &pattern, bool enabled)
+void sinsp_evttype_filter::enable(string &pattern, bool enabled, uint16_t ruleset)
 {
 	regex re(pattern);
 
-	for(auto val : m_evttype_filters)
+	for(auto &val : m_evttype_filters)
 	{
 		if (regex_match(val.first, re))
 		{
-			val.second->enabled = enabled;
+			if(val.second->enabled.size() < (size_t) (ruleset + 1))
+			{
+				val.second->enabled.resize(ruleset + 1);
+			}
+			val.second->enabled[ruleset] = enabled;
+		}
+	}
+}
+
+void sinsp_evttype_filter::enable_tags(set<string> &tags, bool enabled, uint16_t ruleset)
```

Comment:
Mark explained more to me offline. The current behavior works for the falco use case. And yeah, add a comment to clarify.